File: sdk/src/resources/billing.ts
import { BaseResource } from './base';
import type {
  Subscription,
  PricingPlan,
  Invoice,
  ApiResponse,
} from '@escluse/types';

export class BillingResource extends BaseResource {
  async getSubscription(): Promise<ApiResponse<Subscription>> {
    return this.request<ApiResponse<Subscription>>('/billing/subscription');
  }

  async getPlans(): Promise<ApiResponse<PricingPlan[]>> {
    return this.request<ApiResponse<PricingPlan[]>>('/billing/plans');
  }

  async createCheckoutSession(planId: string): Promise<ApiResponse<{ url: string }>> {
    return this.request<ApiResponse<{ url: string }>>('/billing/checkout', {
      method: 'POST',
      body: JSON.stringify({ planId }),
    });
  }

  async cancelSubscription(): Promise<ApiResponse<void>> {
    return this.request<ApiResponse<void>>('/billing/subscription', {
      method: 'DELETE',
    });
  }

  async reactivateSubscription(): Promise<ApiResponse<Subscription>> {
    return this.request<ApiResponse<Subscription>>('/billing/subscription/reactivate', {
      method: 'POST',
    });
  }

  async invoices(): Promise<ApiResponse<Invoice[]>> {
    return this.request<ApiResponse<Invoice[]>>('/billing/invoices');
  }

  async getInvoice(invoiceId: string): Promise<ApiResponse<Invoice>> {
    return this.request<ApiResponse<Invoice>>(`/billing/invoices/${invoiceId}`);
  }
}

File: sdk-python/escluse/resources/billing.py
from typing import Dict, Any
from .base import BaseResource


class BillingResource(BaseResource):
    def get_subscription(self) -> Dict[str, Any]:
        return self._request("GET", "/billing/subscription")

    def get_plans(self) -> Dict[str, Any]:
        return self._request("GET", "/billing/plans")

    def create_checkout_session(self, plan_id: str) -> Dict[str, Any]:
        return self._request("POST", "/billing/checkout", json={"planId": plan_id})

    def cancel_subscription(self) -> Dict[str, Any]:
        return self._request("DELETE", "/billing/subscription")

    def reactivate_subscription(self) -> Dict[str, Any]:
        return self._request("POST", "/billing/subscription/reactivate")

    def invoices(self) -> Dict[str, Any]:
        return self._request("GET", "/billing/invoices")

    def get_invoice(self, invoice_id: str) -> Dict[str, Any]:
        return self._request("GET", f"/billing/invoices/{invoice_id}")